docs/development/contributing/index.md

```diff
@@ -140,9 +191,61 @@ Once all issues are resolved, the maintainer will merge your pull request, and y
 Keep in mind that your work does not have to be perfect right away!
 If you are stuck or unsure about your solution, feel free to open a draft pull request and ask for help.
 
+#### Updating a pull request
+
+Many pull requests go through cycles of repeated reviews and updates before they get merged into the `main` branch of the Polars repository.
+
+To stay up to date with the Polars repository, it is useful to periodically sync your local feature branch with the changes in the remote `main` branch of Polars:
+
+```bash
+git checkout shiny-new-feature
+git fetch upstream
+git rebase upstream/main
+```
+
+If the changes in both branches are incompatible, merge conflicts appear which need to be resolved before you can continue.
+
+Refer to [this guide](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/addressing-merge-conflicts/resolving-a-merge-conflict-using-the-command-line) for how to resolve merge conflicts.
+
+To continue working on your pull request, activate your virtual environment from the root of the Polars repo with:
+
+```bash
+source .venv/bin/activate
+```
+
+When you are finished with making updates to your pull request, follow the steps under "Pushing your changes to GitHub" (see above).
+
+Your changes will appear on your pull request on GitHub and automatically trigger the Continuous Integration pipeline.
+
+## Updating the development environment
+
+For repeatedly contributing to Polars, the development environment needs to be updated periodically.
+
+To keep your fork in sync with the Polars repository run:
+
+```bash
+git checkout main
+git fetch upstream
+git rebase upstream/main
+git push origin main
+```
+
+For updating the project's dependencies and building and installing Python Polars run:
+
+```bash
+make requirements
+make build
```

Comment:
I realised that one should run this, otherwise the polars version is not updated.